### PR TITLE
Use dist constants in rubygems provider for is_omnibus? method

### PR DESCRIPTION
**Pull request by ramereth**
This resolves the following issue [1] which impacts community distributions when using the `gem_package` provider. Instead of using static names, use proper distribution constants which can be easily set by distributions.

TODO: We should update chefdk with Workstation constants once those are created.

[1] https://gitlab.com/cinc-project/distribution/client/-/issues/31

Signed-off-by: Lance Albertson <lance@osuosl.org>

### CODE REVIEW

**Comment by tas50** [comment]
@ramereth There's a spec that needs updating for Workstation as well.

File: lib/chef/provider/package/rubygems.rb [hunk 1]
```diff
@@ -421,11 +421,11 @@ def initialize(new_resource, run_context = nil)
         end
 
         def is_omnibus?
-          if %r{/(opscode|chef|chefdk)/embedded/bin}.match?(RbConfig::CONFIG["bindir"])
+          if %r{/(#{ChefUtils::Dist::Org::LEGACY_CONF_DIR}|#{ChefUtils::Dist::Infra::SHORT}|chef-workstation)/embedded/bin}.match?(RbConfig::CONFIG["bindir"])
```
Comment (tas50) [inline lib/chef/provider/package/rubygems.rb:424]: Is there a dist value for the chef-workstation name?

Comment (ramereth) [reply lib/chef/provider/package/rubygems.rb:424]: Not yet. Would you like me to add that in this PR?

Comment (tas50) [reply lib/chef/provider/package/rubygems.rb:424]: I assume you'd want that for cinc right?

Comment (ramereth) [reply lib/chef/provider/package/rubygems.rb:424]: Yup, let me look and see if there are any other constants we might want to use and get it in.

**Pull request merged**